crates/burn-train/src/metric/confusion_matrix.rs

```diff
@@ -0,0 +1,38 @@
+use burn_core::prelude::{Backend, Bool, Tensor};
+
+#[derive(Clone, Debug)]
+pub struct ConfusionMatrix<B: Backend> {
+    pub true_positive: Tensor<B, 2, Bool>,
+    pub false_positive: Tensor<B, 2, Bool>,
+    pub true_negative: Tensor<B, 2, Bool>,
+    pub false_negative: Tensor<B, 2, Bool>,
+}
+
+impl<B: Backend> ConfusionMatrix<B> {
+    pub fn from(predictions: Tensor<B, 2, Bool>, targets: Tensor<B, 2, Bool>) -> Self {
+        let stats_tensor = predictions.int() + targets.int() * 2;
+        Self {
+            true_positive: stats_tensor.clone().equal_elem(3),
```

Comment:
hey there, I hope everyone doesn't mind me commenting. I am newish to Rust trying to learn and was also working on this so thought it'd be a good learning opportunity for me.

Wouldn't this only work in the binary classification and not multi classification case? I see that in `crates/burn-train/src/metric/precision.rs` line 76 you're using `.greater_elem` and from the code it looks like that just returns a boolean true/false -> 1/0. Given that it doesn't appear that it is possible for the prediction to ever be > 1. Or am I misunderstanding?